cdap-ui/graphql/resolvers/applicationResolvers.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const merge = require('lodash/merge'),
+  urlHelper = require('../../server/url-helper'),
+  cdapConfigurator = require('../../cdap-config.js'),
+  resolversCommon = require('./resolvers-common.js');
+
+let cdapConfig;
+cdapConfigurator.getCDAPConfig()
+  .then(function (value) {
+    cdapConfig = value;
+  });
+
+const applicationsResolver = {
+  Query: {
+    applications: async (parent, args, context, info) => {
+      const namespace = args.namespace
+      const options = resolversCommon.getGETRequestOptions();
+      options['url'] = urlHelper.constructUrl(cdapConfig, `/v3/namespaces/${namespace}/apps`);
+      context.namespace = namespace
+
+      return await resolversCommon.requestPromiseWrapper(options);
+    }
+  }
+};
+
+const applicationResolver = {
+  Query: {
+    application: async (parent, args, context, info) => {
+      const namespace = args.namespace;
+      const name = args.name;
+      const options = resolversCommon.getGETRequestOptions();
```

Comment:
The only arguments that can be passed, per the schema that I defined, are the namespace in the top level queries. I defined a default value for the argument in case that the query does not provide a value for that argument. The namespace is then passed in the applications query to all subqueries.

I am not sure I understand what you mean by "Is the validation of no args checked by apollo server?"

Reply:
So the check whether a namespace and an app name is passed is done by the apollo server then. That was my question.

Reply:
Do add a TODO comment though. We don't seem to be having logical error handling. What happens if I pass in an unknown app name or namespace?

Reply:
You cannot pass an unknown app name since they come from the top level query and used in subqueries. If you pass an invalid namespace, the endpoint returns an error as a string and the server throws an error that it cannot parse that string, since it was expecting an array of applications

Reply:
Done